tests/test_autonomy/test_chain/test_addresses.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2023 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Verify the on-chain addresses against the defined constants."""
+
+import typing as t
+
+import pytest
+import requests
+
+from autonomy.chain import constants
+from autonomy.chain.config import ChainType
+
+
+ADDRESS_FILE_URL = "https://raw.githubusercontent.com/valory-xyz/autonolas-registries/main/docs/configuration.json"
+CHAIN_SLUGS = {ChainType.ETHEREUM: "mainnet", ChainType.GOERLI: "goerli"}
+
+
+def camel_to_snake_case(string: str) -> str:
+    """Convert camelCase to snake_case."""
+    _string = string[0].lower()
+    for char in string[1:]:
+        if char.islower():
+            _string += char
+        else:
+            _string += "_"
+            _string += char.lower()
+    return _string
+
+
+class TestAddresses:
+    """Test addesses."""
+
+    contracts: t.Dict[str, t.List[t.Dict[str, str]]]
+
+    @classmethod
+    def setup_class(cls) -> None:
+        """Setup test class."""
+        chain_configs = requests.get(url=ADDRESS_FILE_URL).json()
+        cls.contracts = {
+            config["name"]: config["contracts"] for config in chain_configs
+        }
+
+    @pytest.mark.parametrize(
+        argnames="chain",
+        argvalues=(
+            ChainType.ETHEREUM,
+            ChainType.GOERLI,
```

Comment:
What about the other chains?

Reply:
https://github.com/valory-xyz/open-autonomy/pull/2030#discussion_r1318304519